### PR TITLE
style: prefer empty/`!` over `-n`/`-z`

### PR DESCRIPTION
**Pull request by scop**
Originally from comments in https://github.com/scop/bash-completion/pull/678.

Docs only for now; we should implement the actual change in this PR too.

### CODE REVIEW

**Comment by scop** [comment]
Rebased and hopefully completed. This might interfere/conflict with other sweeping changes going on -- if so, we can leave this for later and process more important things first, or sneak this in in the middle, I don't mind :)

File: configure.ac [hunk 1]
```diff
@@ -8,7 +8,7 @@ AC_PROG_LN_S
 AC_PROG_MKDIR_P
 AC_PROG_SED
 AC_ARG_WITH([pytest],[  --with-pytest=executable],[PYTEST="$withval"])
-if test -z "$PYTEST"; then
+if test ! "$PYTEST"; then
```
Comment (akinomyoga) [inline configure.ac:11]:
```suggestion
if test "x$PYTEST" = x; then
```

Not sure if we really need to care about (pre-POSIX) Bourne `sh`, but Autotools seem to target Bourne `sh`. I think it is completely fine to forget about the Bourne `sh`, but I just noticed this.

I have checked the behavior of Bourne `sh`, and it seems to fail (i.e., output an error message) with `test ! "$PYTEST"` when `$PYTEST` contains a unary operator like `-f`. On the other hand, `test -z "$PYTEST"` seems to work always as expected at least in the Bourne `sh` that I tested. I guess the Bourne `sh` way is `test "x$PYTEST" = x`.

**Review by akinomyoga — APPROVED** [review]
Thank you! I guess the conflict resolution for this is not that difficult, so we could just merge this.

File: configure.ac (hunk 1 above)
Comment (scop) [reply configure.ac:11]: Yeah, I actually thought I'd leave this one alone, but reading the POSIX spec made me think it'd work and I failed to think about the case the variable contains an operator. Thanks for pointing it out!

**Comment by akinomyoga** [comment]
Thanks! I'm fine with merging this PR. You can merge it any time now or later when you think it is useful.

File: bash_completion [hunk 2]
```diff
@@ -1989,7 +1989,7 @@ _cd()
 
     # Use standard dir completion if no CDPATH or parameter starts with /,
     # ./ or ../
-    if [[ -z ${CDPATH:-} || $cur == ?(.)?(.)/* ]]; then
+    if [[ ${CDPATH:-} || $cur == ?(.)?(.)/* ]]; then
```
Comment (akinomyoga) [inline bash_completion:1992]: The test for `cd` is failing... Ah, this is a typo; `!` seems to have been missing here.

**Review by akinomyoga — CHANGES_REQUESTED** [review]

File: bash_completion (hunk 2 above)
Comment (scop) [reply bash_completion]: Ah! Fixed, merging.

**Pull request merged**